src/test/java/com/microsoft/sqlserver/jdbc/DatabaseMetadataTest.java

```diff
@@ -0,0 +1,88 @@
+package com.microsoft.sqlserver.jdbc;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.AbstractSQLGenerator;
+
+public class DatabaseMetadataTest extends AbstractTest {
+
+    private static String tableName = AbstractSQLGenerator.escapeIdentifier("DBMetadataTestTable");
+    private static String col1Name = AbstractSQLGenerator.escapeIdentifier("p1");
+    private static String col2Name = AbstractSQLGenerator.escapeIdentifier("p2");
+    private static String col3Name = AbstractSQLGenerator.escapeIdentifier("p3");
+
+    @BeforeAll
+    public static void setupTests() throws Exception {
+        setConnection();
+    }
+
+    @Test
+    public void testGetIndexInfo() throws SQLException, SQLServerException {
+        ResultSet rs = null;
+        try (Connection connection = getConnection(); Statement stmt = connection.createStatement()) {
+        	TestUtils.dropTableIfExists(tableName, stmt);
+        	String createTableSQL = "CREATE TABLE " + tableName + " (" +
```

Comment:
Let us use below approach with init and terminate methods for table creation and doppping as it provides idempotency across repeated test runs

        @BeforeEach
        public void init() throws Exception {
            try (Connection con = getConnection()) {
                con.setAutoCommit(false);
                try (Statement stmt = con.createStatement()) {
                    TestUtils.dropTableIfExists(tableName, stmt);
                    stmt.executeUpdate(
                            "CREATE TABLE " + tableName + " (col1 INT, col2 VARCHAR(" + TEST_STRING.length() + "))");
                }
                con.commit();
            }
        }



       @AfterEach
        public void terminate() throws Exception {
            try (Connection con = getConnection(); Statement stmt = con.createStatement()) {
                try {
                    TestUtils.dropTableIfExists(tableName, stmt);
                } catch (SQLException e) {
                    fail(TestResource.getResource("R_unexpectedException") + e.getMessage());
                }
            }
        }